File: 2366-minimum-replacements-to-sort-the-array/2366-minimum-replacements-to-sort-the-array.cpp
using LL = long long;
class Solution {
public:
    long long minimumReplacement(vector<int>& nums) {
        LL ret = 0;
        for(int i = nums.size() - 2; i >= 0; --i) {
            int y = nums[i]; int x = nums[i+1];
            if(y <= x) continue;
            if(y%x == 0) {
                ret += y/x-1;
                nums[i] = x;
            } else {
                ret += y/x;
                nums[i] = y/(y/x+1);
            }
        }
        return ret;
    }
};

File: 2366-minimum-replacements-to-sort-the-array/2366-minimum-replacements-to-sort-the-array.rb
# @param {Integer[]} nums
# @return {Integer}
def minimum_replacement(nums)
    ret = 0
    i = nums.size - 2
    while i >= 0 do
        y = nums[i]
        x = nums[i+1]
        if y <= x
            i -= 1
            next
        end
        if y%x == 0
            ret += y/x - 1
            nums[i] = x
        else
            ret += y/x
            nums[i] = y/(y/x + 1)
        end
        i -= 1
    end
    ret

end